sxs/waveforms/memory.py

```diff
@@ -249,13 +250,25 @@ def add_memory(h, integration_start_time=None):
     integration_start_time : float, optional
         Time at which the energy flux integral should begin.  The default is
         `h.t[0]`.
+    psi4 : WaveformModes, optional
+        WaveformModes object corresponding to psi4
 
     Returns
     -------
-    h_with_memory : WaveformModes
-        WaveformModes object corresponding to the strain with electric memory
+    (h_with_memory, psi4_with_memory) : tuple of WaveformModes
+       WaveformModes objects corresponding to the strain with electric memory
+       and psi4 with electric memory.  If psi4 is None, then psi4_with_memory
+       is also None.
 
     """
-    h_with_memory = WaveformModes(MTS(h) + J_E(h, integration_start_time=integration_start_time))
-    h_with_memory.register_modification(add_memory, integration_start_time=integration_start_time)
-    return h_with_memory
+    h_memory_correction = J_E(h, integration_start_time=integration_start_time)
+    h_with_memory = WaveformModes(MTS(h) + h_memory_correction)
```

Comment:
Do we want to copy over the metadata as well? E.g.,
`h_with_memory.metadata = h.metadata`

Reply:
Note the difference between `metadata` and `_metadata`.  The latter contains all the info like the time array, `ell_min`, `ell_max`, data type, etc., and is copied over automatically already.  On the other hand `h.metadata` might not even be present; it contains information from the metadata.json/txt file, if it happened to be in the same directory when the waveform was loaded.  It's more for interactive convenience than correctness, so I think it should be dropped even if it's present in the original waveform.

Reply:
Ah ok I see, thanks for the clarification